src/Microsoft.IdentityModel.Tokens/Validation/IssuerValidationResult.cs

```diff
@@ -11,17 +11,26 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     internal class IssuerValidationResult : ValidationResult
     {
+        internal enum ValidationSource
+        {
+            NotValidated = 0,
+            IssuerIsConfigurationIssuer,
+            IssuerIsValidIssuer,
+            IssuerIsAmongValidIssuers
+        }
+
         private Exception _exception;
 
         /// <summary>
         /// Creates an instance of <see cref="IssuerValidationResult"/>
         /// </summary>
         /// <paramref name="issuer"/> is the issuer that was validated successfully.
-        public IssuerValidationResult(string issuer)
+        public IssuerValidationResult(string issuer, ValidationSource source)
```

Comment:
Is this intended to be a breaking change? Since this is a public method we usually overload if we want to alter the args

Reply:
I ignored for a moment that this is public and can be implemented by others as a delegate.
Thanks for pointing this out.

Reply:
Added a default value and updated the documentation. Not sure however if we want to add extra values for others implementing their own delegates or we just use this when relying on the internal implementation.